tests/src/system/basic/WskActionTests.scala

```diff
@@ -0,0 +1,313 @@
+        (wp, assetHelper) =>
+            val name1 = "action1"
+            val name2 = "action2"
+            val names = Seq(name1, name2)
+
+            assetHelper.withCleaner(wsk.action, name1) {
+                (action, _) => action.create(name1, Some(TestUtils.getCatalogFilename("samples/hello.js")))
+            }
+            assetHelper.withCleaner(wsk.action, name2) {
+                (action, _) => action.create(name2, Some(TestUtils.getCatalogFilename("samples/wc.js")))
+            }
+
+            val strings = Seq("i have three", "i have four words", "i have five words, period!")
+
+            val activations = for {
+                string <- strings
+                name <- names
+            } yield (wsk.action.invoke(name, Map("payload" -> string.toJson)), string)
+
+            activations.par.foreach {
+                case (rr, testString) =>
+                    withActivation(wsk.activation, rr) {
+                        activation =>
+                            activation.response.status shouldBe "success"
+                            activation.logs.get.mkString(" ") should include(testString)
+                    }
+            }
+    }
+
+    it should "blocking invoke nested blocking actions" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "nestedBlockingAction"
+            val child = "wc"
+
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getCatalogFilename("samples/wcbin.js")))
+            }
+            assetHelper.withCleaner(wsk.action, child) {
+                (action, _) => action.create(child, Some(TestUtils.getCatalogFilename("samples/wc.js")))
+            }
+
+            val run = wsk.action.invoke(name, Map("payload" -> testString.toJson), blocking = true)
+            val activation = wsk.parseJsonString(run.stdout).convertTo[CliActivation]
+
+            val wordCount = testString.split(" ").length
+            activation.response.result.get shouldBe JsObject("binaryCount" -> s"${wordCount.toBinaryString} (base 2)".toJson)
+
+            println(activation)
+    }
+
+    it should "blocking invoke an asynchronous action" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "helloAsync"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getCatalogFilename("samples/helloAsync.js")))
+            }
+
+            val run = wsk.action.invoke(name, Map("payload" -> testString.toJson), blocking = true)
+
+            val activation = wsk.parseJsonString(run.stdout).convertTo[CliActivation]
+
+            activation.response.status shouldBe "success"
+            activation.response.result shouldBe Some(testResult)
+            activation.logs shouldBe None
+    }
+
+    it should "invoke an asynchronous action" in withAssetCleaner(wskprops) {
```

Comment:
same as:
` +    it should "invoke an action with a space in the name" in withAssetCleaner(wskprops) { `
?